plugins/destination/s3/client/write.go

```diff
@@ -14,10 +14,15 @@ import (
 	"github.com/google/uuid"
 )
 
+const (
+	PathVarTable = "{{TABLE}}"
+	PathVarUUID  = "{{UUID}}"
+)
+
 func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data [][]any) error {
-	name := fmt.Sprintf("%s/%s.%s.%s", c.pluginSpec.Path, table.Name, c.pluginSpec.Format, uuid.NewString())
-	if c.pluginSpec.NoRotate {
-		name = fmt.Sprintf("%s/%s.%s", c.pluginSpec.Path, table.Name, c.pluginSpec.Format)
+	name := strings.ReplaceAll(c.pluginSpec.Path, PathVarTable, table.Name)
+	if !c.pluginSpec.NoRotate {
```

Comment:
no need to check, just update, as the set defaults + validate already did their job